sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -3130,10 +3130,18 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase {
       messageParameters = Map.empty)
   }
 
-  def invalidTimestampExprForTimeTravel(expr: Expression): Throwable = {
-    new AnalysisException(
-      errorClass = "_LEGACY_ERROR_TEMP_1335",
-      messageParameters = Map("expr" -> expr.sql))
+  def invalidTimestampExprForTimeTravel(
+      errorClass: String,
+      expr: Expression,
+      dataType: Option[DataType] = None): Throwable = {
+    dataType match {
+      case Some(v) =>
+        new AnalysisException(errorClass = errorClass,
+          messageParameters = Map("expr" -> toSQLExpr(expr), "dataType" -> toSQLType(v)))
+      case _ =>
+        new AnalysisException(errorClass = errorClass,
+          messageParameters = Map("expr" -> toSQLExpr(expr)))
+    }
   }
```

Comment:
Could you make more readable and just overload the function:
```suggestion
  def invalidTimestampExprForTimeTravel(errorClass: String, expr: Expression): Throwable = {
      new AnalysisException(
        errorClass = errorClass,
        messageParameters = Map("expr" -> toSQLExpr(expr)))
  }

  def invalidTimestampExprForTimeTravel(
      errorClass: String,
      expr: Expression,
      dataType: DataType): Throwable = {
    new AnalysisException(
      errorClass = errorClass,
      messageParameters = Map(
        "expr" -> toSQLExpr(expr),
        "dataType" -> toSQLType(dataType)))
  }
```